CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+# Contributing to Mun
+
+Ah! Good to see you are reading this. We feel that developing a programming
+language without the input of our target audience is counterproductive, therefor
+we gladly welcome contributions from all developers! 
+
+If you haven't already, [join our discord server](https://discord.gg/SfvvcCU). We can help you get underway with working on the things you're most excited about.
+
+Here are some important resources to get you on your way:
+
+- [Our roadmap](https://trello.com/b/ZcMiREnC/mun-roadmap) gives a very broad
+  overview of where we're heading.
+- We're usually online on our [our discord server](https://discord.gg/SfvvcCU)
+  during business hours.
+- Bugs? [Report them on GitHub](https://github.com/mun-lang/mun/issues)
+- What is our vision on [testing](testing)?
+  - Try to use [Test Driven Development](#test-driven-development)
+  - About [Snapshot tests](#snapshot-tests)
+- How to [submit changes](#submitting-changes)?
+  - Writing [commit messages](#commit-message-format)
+
+## Testing
+
+We feel that having an extensive [regression
+testing](https://en.wikipedia.org/wiki/Regression_testing) suite allows
+developers to add new features or change existing features with the confidence
+that their changes behave the way they expect it to behave without unknowingly
+impacting other features.
+
+### Test Driven Development
+
+We try to implement new features using [Test Driven Development
+(TTD)](https://en.wikipedia.org/wiki/Test-driven_development). In practice this
+means that we implement new features by first writing a test to ensure the
+feature works. This allows us to very easily integrate regression testing
+because all the code we used to test a feature still exists.
+
+### Snapshot tests
+
+We often use snapshot tests to test whether a feature generates (and keeps
+generating) the correct output. A snapshot test asserts that the output of an
+operation doesn't change. The [insta crate](https://crates.io/crates/insta) is
+used throughout the codebase to enable this functionality. See
+[crates/mun_hir/src/ty/tests.rs](crates/mun_hir/src/ty/tests.rs) for an example.
+
+## Submitting changes
+
+Please send a [GitHub Pull Request to
+mun-lang/mun](https://github.com/mun-lang/mun/pull/new/master) with a clear list
+of what you've done (read more about [pull
+requests](http://help.github.com/pull-requests/)). When you send a pull request
+try to make sure you include tests that validate the feature you implemented
+([read about testing in Mun](testing)). Also, make sure you run `cargo fmt` to
```

Comment:
If `cargo-husky already does this automatically, do we still need to mention this separately?

Reply:
Yes because you can totally ignore all that. I also want to make sure we document what is required for a PR even though we automated it.